src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -333,7 +341,14 @@ private void SetStartHelper(Delegate start, int maxStackSize)
         public bool IsBackground
         {
             get => IsBackgroundNative();
-            set => SetBackgroundNative(value);
+            set
+            {
+                SetBackgroundNative(value);
+                if (!value && !_mayNeedResetForThreadPool)
```

Comment:
```suggestion
                if (!value)
```

I would not worry about trying to avoid the extra write.